CMakeLists.txt

```diff
@@ -1,23 +1,47 @@
 cmake_minimum_required(VERSION 3.13.4)
-project(CMaize_Public_Depend VERSION 1.0.0 LANGUAGES CXX)
+project(CMaizePublicDepend VERSION 1.0.0 LANGUAGES CXX)
 
 include("${PROJECT_SOURCE_DIR}/cmake/get_cpp.cmake")
+include(GNUInstallDirs)
 
 cpp_find_or_build_dependency(
     CMakePublic
-    #used for testing & dev -- URL github.com/ally3999/CMakePublic
-    URL github.com/CMaizeExamples/CMakePublic
+    #used for testing & dev
+    URL github.com/ally3999/CMakePublic
+    #URL github.com/CMaizeExamples/CMakePublic
     BUILD_TARGET CMakePublic
     FIND_TARGET CMakePublic::CMakePublic
 )
 
+#debugging message
+#message("CMakePublic_DIR: ${CMakePublic_DIR}")
+
 cpp_add_library(
-    CMaizePublicDepend
+    ${PROJECT_NAME}
     SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}"
     INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}"
     DEPENDS CMakePublic
 )
 
-target_include_directories (CMaizePublicDepend PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+#do I need to modify the target_include_directories call in this file (or possibly in CMakePublic?)
+#I keep getting this error:
+#/Users/ally/CMaizeExamples/CMaizePublicDepend/cmake_public_depend.cpp:2:10: fatal error: 
+#      'cmake_public.hpp' file not found
+##include <cmake_public.hpp>
+target_include_directories (${PROJECT_NAME} PUBLIC 
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+)
```

Comment:
Trying to determine if/how I need to modify the target_include_directories call in this file (or possibly in CMakePublic)? I keep getting this error:
/Users/ally/CMaizeExamples/CMaizePublicDepend/cmake_public_depend.cpp:2:10: fatal error: 
      'cmake_public.hpp' file not found
#include <cmake_public.hpp>
It also might just be something really obvious I'm missing/forgetting about header files, but I thought since I included cmake_public.hpp with the CMakePublic target that any downstream targets would also include it? Will keep investigating this.

Reply:
When debugging build errors try `VERBOSE=1 <the build command>`. That'll spit out the full commands being run (from which you could see what include paths are being used).

I "think" this is caused by your change from `CMAKE_CURRENT_SOURCE_DIR` to `CMAKE_SOURCE_DIR`. When CMaizePublicDepend includes CMakePublic I believe `CMAKE_SOURCE_DIR` is CMaizePublicDepend not CMakePublic. `CMAKE_CURRENT_SOURCE_DIR` is always the directory containing that file, where as `CMAKE_SOURCE_DIR` is the directory containing the source file being executed. They are usually the same unless a source file is included in another via `include()`.